src/plugins/intel_cpu/src/nodes/executors/acl/acl_pooling.cpp

```diff
@@ -0,0 +1,182 @@
+    unsigned int kernel_w = (poolingAttrs.kernel.size() >= 2) ? poolingAttrs.kernel[1] : poolingAttrs.kernel[0];
+    unsigned int kernel_h = (poolingAttrs.kernel.size() >= 2) ? poolingAttrs.kernel[0] : 1;
+    unsigned int stride_x = (poolingAttrs.stride.size() >= 2) ? poolingAttrs.stride[1] : poolingAttrs.stride[0];
+    unsigned int stride_y = (poolingAttrs.stride.size() >= 2) ? poolingAttrs.stride[0] : 1;
+
+    PoolingType pool_type;
+    bool exclude_padding = false;
+    if (poolingAttrs.algorithm == Algorithm::PoolingMax) {
+        pool_type = PoolingType::MAX;
+        exclude_padding = (poolingAttrs.pad_type != op::PadType::EXPLICIT);
+    } else if (poolingAttrs.algorithm == Algorithm::PoolingAvg) {
+        pool_type = PoolingType::AVG;
+        exclude_padding = poolingAttrs.exclude_pad;
+    } else {
+        DEBUG_LOG("Unknown pooling algorithm: ", static_cast<int>(poolingAttrs.algorithm));
+        return false;
+    }
+    DimensionRoundingType round = (poolingAttrs.rounding == op::RoundingType::CEIL) ?
+                                   DimensionRoundingType::CEIL : DimensionRoundingType::FLOOR;
+
+    if (srcDimsSize == 5) {
+        if (dstDescsSize > 1) {
+            DEBUG_LOG("NEPooling3dLayer does not support indices");
+            return false;
+        } else {
+            unsigned int kernel_d  = poolingAttrs.kernel[2];
+            unsigned int stride_z  = poolingAttrs.stride[2];
+            unsigned int pad_front = poolingAttrs.data_pad_begin[2];
+            unsigned int pad_back  = poolingAttrs.data_pad_end[2];
+            pool3d_info->pool_type       = pool_type;
+            pool3d_info->exclude_padding = exclude_padding;
+            pool3d_info->pool_size       = arm_compute::Size3D(kernel_w, kernel_h, kernel_d);
+            pool3d_info->stride          = arm_compute::Size3D(stride_x, stride_y, stride_z);
+            pool3d_info->padding         = arm_compute::Padding3D(pad_left, pad_right, pad_top, pad_bottom, pad_front, pad_back);
+            pool3d_info->round_type      = round;
+            arm_compute::Status s = arm_compute::NEPooling3dLayer::validate(&srcTensorInfo, &dstTensorInfo, *pool3d_info);
+            if (!s) {
+                DEBUG_LOG("NEPooling3dLayer validation failed: ", s.error_description());
+                return false;
+            }
+        }
+    } else {
+        pool_info->data_layout       = dataLayout;
+        pool_info->pool_size         = arm_compute::Size2D(kernel_w, kernel_h);
+        pool_info->pad_stride_info   = arm_compute::PadStrideInfo(stride_x, stride_y, pad_left, pad_right, pad_top, pad_bottom, round);
+        pool_info->pool_type         = pool_type;
+        pool_info->exclude_padding   = exclude_padding;
+        if (dstDescsSize > 1) {
+            TensorInfo indTensorInfo = TensorInfo(shapeCast(*indDims), 1, arm_compute::DataType::U32, dataLayout);
```

Comment:
Seems like you also have to check that indices output has U32 precision inside AclPoolingExecutorBuilder::isSupported

Reply:
Added